code/game/machinery/hybrsia_lights.dm

```diff
@@ -0,0 +1,457 @@
+/obj/structure/machinery/colony_floodlight/street
+	name = "Colony Streetlight"
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "street_off"
+	layer = ABOVE_XENO_LAYER
+
+/obj/structure/machinery/colony_floodlight/street/update_icon()
+	if(damaged)
+		icon_state = "street_dmg"
+	else if(is_lit)
+		icon_state = "street_on"
+	else
+		icon_state = "street_off"
+
+// Traffic
+
+/obj/structure/machinery/colony_floodlight/traffic
+	lum_value = 0
+	name = "traffic light"
+	desc = "A traffic light"
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "trafficlight"
+	bound_width = 32
+	bound_height = 32
+	density = TRUE
+	health = 200
+	layer = ABOVE_XENO_LAYER
+
+/obj/structure/machinery/colony_floodlight/traffic/update_icon()
+	if(damaged)
+		icon_state = "trafficlight_damaged"
+	else if(is_lit)
+		icon_state = "trafficlight_on"
+	else
+		icon_state = "trafficlight"
+
+/obj/structure/machinery/colony_floodlight/traffic_alt
+	lum_value = 0
+	name = "traffic light"
+	desc = "A traffic light"
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "trafficlight_alt"
+	bound_width = 32
+	bound_height = 32
+	density = TRUE
+	health = 200
+	layer = ABOVE_XENO_LAYER
+
+/obj/structure/machinery/colony_floodlight/traffic_alt/update_icon()
+	if(damaged)
+		icon_state = "trafficlight_alt_damaged"
+	else if(is_lit)
+		icon_state = "trafficlight_alt_on"
+	else
+		icon_state = "trafficlight_alt"
+
+// Engineer Floor lights
+
+/obj/structure/machinery/engineerconsole_switch
+	name = "Giant Alien Console"
+	icon = 'icons/obj/structures/props/64x64_hybrisarandomprops.dmi'
+	icon_state = "engineerconsole"
+	desc = "A Giant Alien console of some kind, unlike anything you've ever seen before. Who knows the purpose of this strange technology..."
+	bound_height = 32
+	bound_width = 32
+	unslashable = TRUE
+	unacidable = TRUE
+	indestructible = TRUE
+	density = TRUE
+	anchored = TRUE
+	var/ispowered = TRUE
+	var/turned_on = 0 //has to be toggled in engineering
+	use_power = USE_POWER_NONE
+	var/list/floodlist = list() // This will save our list of floodlights on the map
+	power_machine = TRUE
+
+/obj/structure/machinery/engineerconsole_switch/Initialize(mapload, ...)
+	. = ..()
+	return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/machinery/engineerconsole_switch/LateInitialize()
+	. = ..()
+	for(var/obj/structure/machinery/colony_floodlight/engineer_circular/F in GLOB.ship_floodlights)
+		floodlist += F
+		F.fswitch = src
+	start_processing()
+
+/obj/structure/machinery/engineerconsole_switch/Destroy()
+	for(var/obj/structure/machinery/colony_floodlight/engineer_circular/floodlight as anything in floodlist)
+		floodlight.fswitch = null
+	floodlist = null
+	return ..()
+
+
+/obj/structure/machinery/engineerconsole_switch/update_icon()
+	if(!ispowered)
+		icon_state = "engineerconsole"
+	else if(turned_on)
+		icon_state = "engineerconsole"
+	else
+		icon_state = "engineerconsole"
```

Comment:
This doesn't do anything.

Reply:
Yeah it's not supposed to have different icon states like the "colony floodlight switch". I wasn't sure if this was needed or not, is it safe to delete it?

Reply:
Yes you can just delete the entire `/obj/structure/machinery/engineerconsole_switch/update_icon()` block